lib/common/patchset.c

```diff
@@ -539,13 +539,9 @@ xml_log_patchset(uint8_t log_level, const char *function, xmlNode *patchset)
 
             } else if (strcmp(op, "modify") == 0) {
                 xmlNode *clist = first_named_child(change, XML_DIFF_LIST);
-                char buffer_set[PCMK__BUFFER_SIZE];
-                char buffer_unset[PCMK__BUFFER_SIZE];
-                int o_set = 0;
-                int o_unset = 0;
+                GString *buffer_set = g_string_sized_new(256);
+                GString *buffer_unset = g_string_sized_new(256);
```

Comment:
It would be more efficient to take these variables out of the loop, allocate them here if they're null otherwise truncate them, and free them after the loop.